### PR TITLE
Protected Routes

### PR DESCRIPTION
**Pull request by noarrassam**
### What this PR does (required):
- list PR functionality here
1. Created Protected Routes inside client folder.
2. Imported PrivateRoute inside App.tsx and used it for dashboard.

### Screenshots / Videos (required):
- 
Screen Shot 2021-08-10 at 12 35 22 PM

Screen Shot 2021-08-10 at 12 35 13 PM

Screen Shot 2021-08-10 at 12 35 31 PM

Screen Shot 2021-08-10 at 12 35 42 PM

Screen Shot 2021-08-10 at 12 36 16 PM

### Any information needed to test this feature (required):
N/A

### Any issues with the current functionality (optional):
N/A

### CODE REVIEW

File: client/src/components/routing/PrivateRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React, { Fragment, useContext } from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { AuthContext } from '../../context/useAuthContext';
+
+const PrivateRoute: React.FunctionComponent<any> = ({ component: Component, ...rest }) => {
+  const authContext = useContext(AuthContext);
+  const { loggedInUser } = authContext;
```
Comment (ugur-akin) [inline client/src/components/routing/PrivateRoute.tsx:7]: We can probably combine these two lines without any drawbacks.

**Review by ugur-akin — COMMENTED** [review]
Nice changes! One thing though, I would recommend writing more descriptive commit messages to create meaningful commits - it would very helpful to both yourself and your collaborators!

File: client/src/App.tsx [hunk 2]
```diff
@@ -20,12 +21,7 @@ function App(): JSX.Element {
               <Switch>
                 <Route exact path="/login" component={Login} />
                 <Route exact path="/signup" component={Signup} />
-                <Route exact path="/dashboard">
-                  <Dashboard />
-                </Route>
-                <Route path="*">
-                  <Redirect to="/login" />
-                </Route>
+                <PrivateRoute path="/dashboard" component={Dashboard} />
```
Comment (rajivtitus) [inline client/src/App.tsx:24]: Just a suggestion, wouldn't it be better to use the exact param to disable partial matching for a route? :)

**Review by rajivtitus — COMMENTED** [review]
Hey @michaeldotdev and @jedchernandez, No rush but whenever possible can either of you take care of the minor comments and complete this feature. Thank you :)

**Comment by rajivtitus** [comment]
Closing this PR as this feature has been completed in https://github.com/hatchways/team-boolean-autocrats/pull/54

**Pull request closed without merging**